ui/src/wizard/StorageAccountListStep.ts

```diff
@@ -94,22 +102,48 @@ export class StorageAccountListStep<T extends IStorageAccountWizardContext> exte
         return wizardContext;
     }
 
-    private async getQuickPicks(storageAccountsTask: Promise<StorageAccount[]>): Promise<IAzureQuickPickItem<StorageAccount | undefined>[]> {
-        const picks: IAzureQuickPickItem<StorageAccount | undefined>[] = [{
+    private async getQuickPicks(storageAccountsTask: Promise<StorageAccount[]>): Promise<IAzureQuickPickItem<StorageAccount | undefined | string>[]> {
+        const picks: IAzureQuickPickItem<StorageAccount | undefined | string>[] = [{
             label: localize('NewStorageAccount', '$(plus) Create new storage account'),
             description: '',
             data: undefined
         }];
 
+        const kindRegExp: RegExp = new RegExp(`^${convertFilterToPattern(this._filterOptions.kind)}$`, 'i');
+        const skuRegExp: RegExp = new RegExp(`^${convertFilterToPattern(this._filterOptions.performance)}_${convertFilterToPattern(this._filterOptions.replication)}$`, 'i');
+
+        let hasFilteredAccounts: boolean = false;
         const storageAccounts: StorageAccount[] = await storageAccountsTask;
-        return picks.concat(storageAccounts.map((sa: StorageAccount) => {
-            return {
+        for (const sa of storageAccounts) {
+            // tslint:disable:strict-boolean-expressions
+            if (!sa.kind || !sa.kind.match(kindRegExp) || !sa.sku || !sa.sku.name.match(skuRegExp)) {
```

Comment:
Isn't this more of a 'select' rather than a 'filter'?  If I'm understanding this correctly, the filterOptions are actually selecting which accounts CAN be used rather than filtering them out?

Reply:
Mmm good point. I'll keep 'filter' in the name, but flip the logic. I think it makes more sense to specify which accounts _can't_ be used rather than _can_ be used.